From 7ece40470f6f19bb49b5da5ed652821a8f6fbf97 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Thorsten=20Wi=C3=9Fmann?= <edu@thorsten-wissmann.de>
Date: Mon, 3 May 2021 12:51:48 +0200
Subject: [PATCH] Include <limits> for std::numeric_limits (#1305)

This fixes a compilation error on Gentoo with GCC 11[1], complaining
about an undefined std::numeric_limits in the namespace std. According
to cppreference std::numeric_limits requires including <limits>. This
adds the include to floating.cpp, where the error occurred, and also to
rectangle.cpp (which also uses std::numeric_limits but did not have the
include).

[1] https://bugs.gentoo.org/787848
---
 src/floating.cpp  | 1 +
 src/rectangle.cpp | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/floating.cpp b/src/floating.cpp
index 33b4f0cdaf5a..9a86ee4777bf 100644
--- a/src/floating.cpp
+++ b/src/floating.cpp
@@ -3,6 +3,7 @@
 #include <algorithm>
 #include <climits>
 #include <cstdlib>
+#include <limits>
 #include <unordered_set>
 
 #include "client.h"
diff --git a/src/rectangle.cpp b/src/rectangle.cpp
index 4bcdb334f852..5b474e535bd3 100644
--- a/src/rectangle.cpp
+++ b/src/rectangle.cpp
@@ -1,6 +1,7 @@
 #include "rectangle.h"
 
 #include <X11/Xutil.h>
+#include <limits>
 #include <queue>
 
 #include "ipc-protocol.h"
-- 
2.30.2

